contrib/http_jaxrs/src/main/java/io/opencensus/contrib/http/jaxrs/JaxrsContainerFilter.java

```diff
@@ -0,0 +1,132 @@
+ * OpenCensus. Filter will instrument any endpoint marked with the {@link Metrics} annotation.
+ * Filter will also start a scoped span so that child spans may be added.
+ *
+ * @since 0.19
+ */
+@Metrics
+@Provider
+@Priority(Priorities.USER - 100)
+public class JaxrsContainerFilter implements ContainerRequestFilter, ContainerResponseFilter {
+
+  private static final String CONTEXT_PROPERTY = "opencensus.context";
+  private static final String SPAN_PROPERTY = "opencensus.span";
+  private static final Getter<ContainerRequestContext> GETTER =
+      new Getter<ContainerRequestContext>() {
+        @Override
+        public String get(ContainerRequestContext request, String key) {
+          return request.getHeaderString(key);
+        }
+      };
+
+  private final HttpServerHandler<
+          ExtendedContainerRequest, ContainerResponseContext, ContainerRequestContext>
+      handler;
+
+  @SuppressWarnings("initialization.fields.uninitialized") // Will be injected by JAX-RS
+  @Context
+  private ResourceInfo info;
+
+  /**
+   * Default constructor construct new instance with {@link JaxrsContainerExtractor}, {@link
+   * io.opencensus.trace.propagation.PropagationComponent#getTraceContextFormat()} and as public
+   * endpoint.
+   *
+   * @see #JaxrsContainerFilter(HttpExtractor, TextFormat, Boolean)
+   */
+  public JaxrsContainerFilter() {
+    this(
+        new JaxrsContainerExtractor(),
+        Tracing.getPropagationComponent().getTraceContextFormat(),
+        /* publicEndpoint= */ true);
+  }
+
+  /**
+   * Construct instance with custom configuration.
+   *
+   * @param extractor the {@code HttpExtractor} used to extract information from the
+   *     request/response.
+   * @param propagationFormat the {@code TextFormat} used in HTTP propagation.
+   * @param publicEndpoint set to true for publicly accessible HTTP(S) server. If true then incoming
+   *     tracecontext will be added as a link instead of as a parent.
+   */
+  public JaxrsContainerFilter(
+      HttpExtractor<ExtendedContainerRequest, ContainerResponseContext> extractor,
+      TextFormat propagationFormat,
+      Boolean publicEndpoint) {
+    this.handler =
+        new HttpServerHandler<>(
+            Tracing.getTracer(), extractor, propagationFormat, GETTER, publicEndpoint);
+  }
+
+  @Override
+  @SuppressWarnings("MustBeClosedChecker") // Close will happen in response filter method
+  public void filter(ContainerRequestContext requestContext) throws IOException {
+    ExtendedContainerRequest extendedRequest = new ExtendedContainerRequest(requestContext, info);
+    HttpRequestContext context = handler.handleStart(requestContext, extendedRequest);
+    requestContext.setProperty(CONTEXT_PROPERTY, context);
+    if (requestContext.getLength() > 0) {
+      handler.handleMessageReceived(context, requestContext.getLength());
+    }
+    requestContext.setProperty(
+        SPAN_PROPERTY, Tracing.getTracer().withSpan(handler.getSpanFromContext(context)));
```

Comment:
I was wondering too. I was looking at the `http-servlet` module and it startled me a bit that it did not start a scope span since I would expect that you wanted the span from the filter to be parent of any new spans being created in between. Like if this is a service and you do a request to another service you would have to get hold of the `HttpRequestContext` to be able to set up the parent span.

Reply:
Lets change the http-servlet and remove the scoped span here.

Reply:
You think that change should be part of this PR - would be a little bit “out of focus” but on the other hand the change is not that complex, I guess.

Reply:
@rghetia I have been considering this looking into the `HttpServerHandler` class as well as its usage in `OcHttpServletFilter`. While I think the servlet filter should take a similar approach and make sure that the http request is actually processed within a scoped span, I am not sure this should be added to `HttpServerHandler`.

While the JAX-RS filter is pretty straight forward in that it will have both the "request" and "response" filter methods invoked the servlet filter is more complex. The servlet filter is also taking care of requests processed asynchronously and handling that use case I guess it makes sense to leave it out of `HttpServerHandler` and leave it up to the concrete implementation to ensure an active scoped span. With an asynchronous request I guess the `OcHttpServletListener` should be setting up the scope.

So I would argue to leave `HttpServerHandler` as-is for now but maybe (in another PR) change `OcHttpServletFilter` and `OcHttpServletListener` to ensure an active scope when the request is being processed.

Reply:
@langecode sounds good. I opened #1663 to track this.